File: backend/app/routes/audits.py
from __future__ import annotations
from fastapi import APIRouter
from app.models.audit import ProjectAudit
from app.services import audit_service

router = APIRouter(prefix="/api/audits", tags=["audits"])


@router.post("/run", response_model=ProjectAudit, status_code=201)
async def run_audit():
    return audit_service.run_audit()


@router.get("/latest", response_model=ProjectAudit)
async def get_latest_audit():
    return audit_service.get_latest_audit()


@router.get("", response_model=list[ProjectAudit])
async def list_audits():
    return audit_service.list_audits()

File: frontend/src/lib/audits.ts
export type AuditSeverity = "info" | "warning" | "critical";
export type AuditStatus = "unhealthy" | "needs_attention" | "production_ready";

export interface AuditIssue {
  id: string;
  severity: AuditSeverity;
  category: string;
  title: string;
  description: string;
  suggestion: string;
  created_at: string;
}

export interface PipelineSummaryItem {
  category: string;
  status: "healthy" | "warning" | "missing" | "failed";
  detail: string;
}

export interface ProjectAudit {
  id: string;
  score: number;
  status: AuditStatus;
  issues: AuditIssue[];
  strengths: string[];
  recommendations: string[];
  pipeline_summary: PipelineSummaryItem[];
  created_at: string;
}

export async function runAudit(): Promise<ProjectAudit> {
  const res = await fetch("/api/audits/run", { method: "POST" });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error((err as { detail?: string }).detail ?? "Audit failed");
  }
  return res.json();
}

export async function getLatestAudit(): Promise<ProjectAudit | null> {
  const res = await fetch("/api/audits/latest");
  if (res.status === 404) return null;
  if (!res.ok) throw new Error("Failed to fetch latest audit");
  return res.json();
}

export async function listAudits(): Promise<ProjectAudit[]> {
  const res = await fetch("/api/audits");
  if (!res.ok) throw new Error("Failed to fetch audit history");
  return res.json();
}
